### ISSUE TITLE
Some packages must be rebuild on rpi to work properly

### ISSUE THREAD

**Issue by teldra** [void-linux/void-packages#4639]
### System

* xuname:  
Void 4.14.79_1 armv7l  uptodate rDDDFF
* package:  
python3-psutil-5.4.8_1 and package release before
python3-bcrypt-3.1.4_1 and package release before
maybe others

### Expected behavior
thefuck should run
htpassword should import bcrypt

### Actual behavior
since python3-psutil-5.4.8_1 is needed for "thefuck", thefuck stops with
`ImportError: cannot import name '_psutil_linux'`

htpasswd can't import bcrypt (no logs at the moment)

### Steps to reproduce the behavior
for psutil: install thefuck on rpi and try to run it
for bcrypt: install apache and python3-bcrypt on rpi and try to use htpasswd with -B (use bcrypt)

**Comment by Hoshpak** [comment]
I think that's because python3 includes the host architecture used while building in the file name, even though the cross-compiled .so-files are proper arm binaries they are called e.g. `_psutil_posix.cpython-36m-x86_64-linux-gnu.so`. As far as I can see, python2 did not do that so there's no issue there.

Unless python has a mechanism to fix that (which I couldn't find), we would probably have to rename the files after building in the build_styles to match the target architecture.

**Comment by andkem** [comment]
I've run into this problem trying to run the borg package on a Raspberry Pi 3 running aarch64. I've tracked the issue to the same problem Hoshpak is mentioning.

The problem is that `build_ext.py` from distutils gets the library suffix on line 681 using `get_config_var('EXT_SUFFIX')`. This is hard-coded and cannot be overridden by normal means.

**Comment by andkem** [comment]
This is a little patch I wrote to hack around the wrong naming issue. It adds a few lines off hacky python to the top of setup.py when the build is run. In the current format, it doesn't check if the line is already there so it will add it again if the source is built without being re-extracted.

Since this is an ugly hack, it might be better to change the file names after the files have been built. I can however tidy the patch up/add a check if the line already exists if anyone would be interested in this solution.

[Hacky patch to remove Python shared library suffix](http://lysator.liu.se/~kempe/0001-python3-build-style-rename-shared-library-suffix.patch)

I deleted my previous comment, because I thought I had made an error in my patch so it didn't work, sorry.
